apps/ob3/serializers.py

```diff
@@ -0,0 +1,159 @@
+        ret = super().to_representation(instance)
+        """Convert the id to a URL"""
+        ret['id'] = f"https://example.com/issuers/{ret['id']}"
+        return ret
+
+class ImageSerializer(serializers.Serializer):
+    type = serializers.CharField(read_only=True, default="Image")
+    id = serializers.URLField()
+
+class AlignmentSerializer(serializers.Serializer):
+    type = serializers.ListField(
+            child=serializers.CharField(),
+            read_only=True,
+            default=["Alignment"]
+    )
+    targetType = serializers.CharField(
+            source='target_type',
+            read_only=True,
+    )
+    targetName = serializers.CharField(
+            source='target_name',
+    )
+    targetDescription = serializers.CharField(
+            source='target_description',
+            required=False,
+            allow_null=True,
+    )
+    targetUrl = serializers.URLField(
+            source='target_url',
+    )
+    targetCode = serializers.CharField(
+            source='target_code',
+            required=False,
+            allow_null=True,
+    )
+
+    def to_representation(self, instance):
+        ret = super().to_representation(instance)
+
+        # TODO: decide on what to do if framework is missing
+        # TODO: we should really make this an enum with allowed values instead
+        framework = instance.get("target_framework", "")
+        ret["targetType"] = f"ext:{framework}Alignment"
+        return ret
+
+
+class AchievementSerializer(OmitNoneFieldsMixin, serializers.Serializer):
+    OMIT_IF_NONE = ['inLanguage', 'ECTS', 'educationProgramIdentifier', 'participationType', 'alignment']
+
+    id = serializers.URLField()
+    type = serializers.ListField(
+            child=serializers.CharField(),
+            read_only=True,
+            default=["Achievement"]
+    )
+    criteria = serializers.DictField(child=serializers.CharField())
+    description = serializers.CharField()
+    name = serializers.CharField()
+    image = ImageSerializer()
+    inLanguage = serializers.CharField(
+            source='in_language',
+            required=False,
+            allow_null=True,
+    )
+    ECTS = serializers.DecimalField(
+            source='ects',
+            decimal_places=1,
+            max_digits=3, # Up to 99,9 ECTS (in reality, it's up to 10.0, IIRC)
+            coerce_to_string=False,
+    )
+    educationProgramIdentifier = serializers.CharField(
+            source='education_program_identifier',
+            required=False,
+            allow_null=True,
+    )
+    participationType = serializers.CharField(
+            source='participation',
+            required=False,
+            allow_null=True,
+    )
+    alignment = AlignmentSerializer(many=True)
+
+    def to_representation(self, instance):
+        ret = super().to_representation(instance)
+        """Convert the id to a URL"""
+        ret['id'] = f"https://example.com/assertions/{ret['id']}"
```

Comment:
Update to edubadges nl URL? Such as https://demo.edubadges.nl/public/assertions/l0FxH-o2QZ2p4CyDfmOdBw

Reply:
Maar dan linken naar de ENV var versie.